src/wallet/orchard.cpp

```diff
@@ -14,15 +14,19 @@ std::optional<libzcash::OrchardSpendingKey> OrchardWallet::GetSpendingKeyForAddr
 
 std::vector<std::pair<libzcash::OrchardSpendingKey, orchard::SpendInfo>> OrchardWallet::GetSpendInfo(
     const std::vector<OrchardNoteMetadata>& noteMetadata,
-    uint256 anchor) const
+    unsigned int anchorConfirmations,
+    const uint256& anchor) const
```

Comment:
Why do we require `anchor` to be passed in when, for a given `anchorConfirmations`, there is only one value it could have?

Reply:
I continue to pass in `anchor` here as a sanity check; we obtain `anchor` from the coins db, which caches the Merkle frontier for each chain state, and here we verify that against the Merkle root computed from the wallet's `BridgeTree` instance. Previously this check was implicit, because we used the anchor to search for the tree state that we're interested in. Performing that search required recomputing each candidate root of the `BridgeTree`; now we no longer have to do that, but I think it's useful to still compute it once to make certain that we're using the correct Merkle path when making the proofs.